flask_mwoauth/__init__.py

```diff
@@ -6,94 +6,75 @@
 # (C) 2013 Merlijn van Deen <valhallasw@arctus.nl>
 # Licensed under the MIT License // http://opensource.org/licenses/MIT
 #
-
-__version__ = '0.2.54'
-
 import sys
 from future.utils import iteritems
-from future.moves.urllib.parse import urlencode
 from flask import request, session, redirect, url_for, flash, Blueprint
-from flask_oauthlib.client import OAuth, OAuthException
+import mwoauth
+import requests
 from requests.models import Request
+from requests_oauthlib import OAuth1
+
+__version__ = '0.2.54'
 
 
 class MWOAuth(object):
-    def __init__(self,
+    def __init__(self, consumer_key, consumer_secret,
```

Comment:
This is a weird pattern.  consumer_key and consumer_secret are required arguments.  But I agree that this will result in some incompatibilities with argument order (didn't realize that).  I'll change it back to something compatible.